### PR TITLE
Expose auto resource limits FG

### PR DESCRIPTION
**Pull request by fossedihelm**
**What this PR does / why we need it**:
Expose a Feature Gate to configure
auto resource limits on Kubevirt.

Also, added missing FGs in `kubebuilder:default` FG outer fields.
The missing FGs were:
- `deployVmConsoleProxy`
- `disableMDevConfiguration`
- `persistentReservation`
- `enableManagedTenantQuota`

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-31898
```

**Release note**:

```release-note
Expose auto resource limits FG
```

### CODE REVIEW

File: api/v1beta1/hyperconverged_types.go [hunk 1]
```diff
@@ -407,6 +407,19 @@ type HyperConvergedFeatureGates struct {
...
+	// TODO update description to also include cpu limits as well, after 4.14
+
+	// Enable KubeVirt to set automatic limits when they are needed.
+	// If ResourceQuota with set memory limits is associated with a namespace, each pod in that namespace must have memory limits set.
+	// By default, KubeVirt does not set such limits to the virt-launcher pod.
+	// When this feature gate is enabled, KubeVirt will set limits to the virt-launcher pod if they are not set manually
+	// and if a resource quota with memory limits is associated with the creation namespace.
+	// Note: this feature is in Developer Preview.
+	// +optional
+	// +kubebuilder:default=false
```
Comment (tiraboschi) [inline api/v1beta1/hyperconverged_types.go:420]: I'd expect this also on lines 69 and 704.

Comment (fossedihelm) [reply api/v1beta1/hyperconverged_types.go:420]: @tiraboschi Can I ask you why the `persistentReservation` is not there?

Comment (tiraboschi) [reply api/v1beta1/hyperconverged_types.go:420]: Good point, this smells like a bug but our tests are not detecting it.
So the tests are not robust enough or kube builder become smarter and duplicating it is not needed anymore.
Let's investigate.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**